ocs_ci/framework/pytest_customization/ocscilib.py

```diff
@@ -622,14 +622,18 @@ def pytest_runtest_makereport(item, call):
     # we only look at actual failing test calls, not setup/teardown
     if rep.failed and ocsci_config.RUN.get("cli_params").get("collect-logs"):
         test_case_name = item.name
-        ocp_logs_collection = True if rep.when == "call" else False
-        mcg = (
+        ocp_logs_collection = ocsci_config.REPORTING.get("ocp_must_gather_collection")
+        ocs_logs_collection = ocsci_config.REPORTING.get("ocs_must_gather_collection")
+        mcg_logs_collection = (
             True if any(x in item.location[0] for x in ["mcg", "ecosystem"]) else False
         )
         try:
             if not ocsci_config.RUN.get("is_ocp_deployment_failed"):
                 collect_ocs_logs(
-                    dir_name=test_case_name, ocp=ocp_logs_collection, mcg=mcg
+                    dir_name=test_case_name,
+                    ocp=ocp_logs_collection,
```

Comment:
As you set it to False in default config then for failed deployment we will not collect OCP must gather logs.  For investigating deployment issues we need OCP must gather so I see it might be problematic for ecosystem team for deployment failures analysis.

[edit] Ah, I see you are using enable_ocp_mg_collection_on_failure in deployment test so it should be resolved by this.